### ISSUE TITLE
BUG in the new PPO trainer

### ISSUE THREAD

**Issue by TingchenFu** [huggingface/trl#2353]
### System Info

In the init function of the new ppo trainer (renamed from ppo trainer v2), it says

```
 if ref_policy is policy:
            raise ValueError(
                "`policy` and `ref_policy` cannot be the same object. If you want `ref_policy` to be the "
                "same as `policy`, you must mass a copy of it, or `None` if you use peft."
            )
```
But actually, the ref_policy cannot be None. Otherwise, it will lead to an error when later in the init function  remove_dropout_in_model() is called.

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

See the description above 

### Expected behavior

peft support.

### Checklist

- [X] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [X] I have included my system information
- [X] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [X] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [X] Any traceback provided is complete

**Comment by Mrinh212375** [comment]
Hi...I think we need to create a copy of the policy model using create_reference_model() function....is it ??

I'm facing another problem in new PPOTrainer()...according to documentation  we need to pass **'module**' unlike the previous version(HF PreTrainedWrapper ).
How to get the HF pretrainedWrapper models and pass to PPOTrainer() as module

**Comment by ccs96307** [comment]
I'm hopeful that https://github.com/huggingface/trl/pull/2344 will address this issue! :raised_hands:

**Comment by TingchenFu** [comment]
I turned to v0.11 as a bypass🤣

**Comment by ccs96307** [comment]
May I ask which version you are using? 

As far as I know, the current version (0.12.1) does not support `PEFT` for `PPOTrainer` yet. However, if you want to try `PEFT` support, you can consider using `pip install git+https://github.com/huggingface/trl.git` to experiment with it. :muscle:

**Comment by ZhaoningYu1996** [comment]
Thank you for your reply. 
I am using 0.12.1.
I will try pip install git+https://github.com/huggingface/trl.git

**Comment by ZhaoningYu1996** [comment]
> I turned to v0.11 as a bypass🤣

Does 0.11 support PEFT?

**Comment by albertvillanova** [comment]
Hey @TingchenFu, sorry for the very late reply — catching up on backlog. 

This was a real bug at the time you filed it, but PEFT support for the new PPO trainer landed shortly after in #2344. On current `main`, the constructor allows `ref_model=None` when a `peft_config` is passed (see https://github.com/huggingface/trl/blob/51c6d3ca31fb4cc80ff719c0844bbdfcd0feeefb/trl/experimental/ppo/ppo_trainer.py#L447-L452) and the dropout-disable loop is guarded with an `if module is not None` check (https://github.com/huggingface/trl/blob/51c6d3ca31fb4cc80ff719c0844bbdfcd0feeefb/trl/experimental/ppo/ppo_trainer.py#L497-L499). Note that the PPO trainer has been moved to `trl.experimental.ppo`. 

Closing as fixed; happy to re-open if you can repro on a recent version with a minimal script + `trl env` output.
